src/peer_transport/transport.ts

```diff
@@ -0,0 +1,164 @@
+import type { Connection } from '@libp2p/interface-connection'
+import { CreateListenerOptions, DialOptions, Listener, symbol, Transport } from '@libp2p/interface-transport'
+import type { ConnectionHandler, TransportManager, Upgrader } from '@libp2p/interface-transport'
+import { multiaddr, Multiaddr } from '@multiformats/multiaddr'
+import type { IncomingStreamData, Registrar } from '@libp2p/interface-registrar'
+import type { PeerId } from '@libp2p/interface-peer-id'
+import { WebRTCMultiaddrConnection } from '../maconn.js'
+import type { Startable } from '@libp2p/interfaces/startable'
+import { WebRTCPeerListener } from './listener.js'
+import type { PeerStore } from '@libp2p/interface-peer-store'
+import { logger } from '@libp2p/logger'
+import { connect, handleIncomingStream } from './handler.js'
+
+const log = logger('libp2p:webrtc:peer')
+
+export const TRANSPORT = '/webrtc'
+export const PROTOCOL = '/webrtc-signaling/0.0.1'
+export const CODE = 281
+
+export interface WebRTCPeerTransportInit {
+  rtcConfiguration?: RTCConfiguration
+}
+
+export interface WebRTCDirectTransportComponents {
+  peerId: PeerId
+  registrar: Registrar
+  upgrader: Upgrader
+  transportManager: TransportManager
+  peerStore: PeerStore
+}
+
+export class WebRTCDirectTransport implements Transport, Startable {
+  private readonly _started = false
+  private readonly handler?: ConnectionHandler
```

Comment:
who sets this?

Reply:
Remaining from an older implementation. This field is included in `CreateListenerOptions`. It wouldn't work here since all listening is done by the transport itself over the protocol.

Reply:
I pushed a fix.